crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -230,27 +260,19 @@ impl<'src> Parser<'src> {
     ///
     /// Matches the `expression` rule in the [Python grammar].
     ///
-    /// The caller can specify whether starred expression is allowed or not. This
-    /// doesn't affect the parsing of a starred expression as it will be parsed
-    /// nevertheless. But, if it is not allowed, an error is reported.
-    ///
-    /// Use [`Parser::parse_star_expression_or_higher`] with [`AllowNamedExpression::No`]
-    /// if the starred expression is required with a bitwise OR precedence.
-    ///
-    /// NOTE: If you have expressions separated by commas and want to parse them
-    /// individually instead of as a tuple, as done by [`Parser::parse_expression_list`]
-    /// use this function!
+    /// NOTE: If you have expressions separated by commas and want to parse them individually
+    /// instead of as a tuple, as done by [`Parser::parse_expression_list`] use this function.
     ///
     /// [Python grammar]: https://docs.python.org/3/reference/grammar.html
     pub(super) fn parse_conditional_expression_or_higher(
```

Comment:
Nit: The context seems to always be `default` except when called from `parse_named_expression_or_higher` or the list parsing. 

Should we split the method into a `parse_conditional_expression_or_higher` that defaults to `ExpressionContext::default` and have a `parse_conditional_expression_or_higher_impl` that has an optional `ExpressionContext`? It could help making writing new expression parsing easier if the default context is what should be used in all places where `parse_conditional_expression_or_higher` is used (I don't need to think that hard about what context I need to pass)

Reply:
Yeah, I didn't realize that. I just counted them and it seems like so:

  | default | non-default | inherit
-- | -- | -- | --
expression list | 6 | 11 |  
named expression | 5 | 11 |  
conditional expression | 19 | 3 | 1